src/ansi-c/library/string.c

```diff
@@ -563,7 +576,16 @@ inline size_t strlen(const char *s)
   return __CPROVER_zero_string_length(s);
   #else
   __CPROVER_size_t len=0;
-  while(s[len]!=0) len++;
+  while(s[len]!=0) 
+  __CPROVER_loop_invariant (
+        (0 <= len) && (len < __CPROVER_OBJECT_SIZE(s)) &&
+        ((len == 0) ||
```

Comment:
[nit]

Is `len == 0` clause needed?
Wouldn't `0 <= k && k < len ==> ...` vacuously pass anyway without it (`false ==> <anything>`)?

Reply:
> Is `len == 0` clause needed?

You are right, we don't need the `len == 0` clause. The verification still successes after removing it.
> Wouldn't `0 <= k && k < len ==> ...` vacuously pass anyway without it (`false ==> <anything>`)?

The expression will vacuously pass when `len` is always `0`. But it is not the case here: `len` can be greater `0`.

Reply:
Yes, I only meant vacuity in case of `len == 0`.